package-res/ccc/core/base/slidingWindow.js

```diff
@@ -0,0 +1,233 @@
+                    if (dimOptions) var dimComp = dimOptions[dimName];
+                    if(!dimComp || !dimComp.comparer){
+                        dim.type.setComparer(def.ascending); //???
+                    }
+                }, this);
+                
+            }, this);
+
+            this._preserveLayout();
+            this._setFixedRatio();
+
+        },
+
+        _preserveAxisColorMap: function(axis) { axis.setPreserveColorMap(); },
+
+        _preserveLayout: function() { this.chart.options.preserveLayout = true; },
+
+        _setFixedRatio: function(){
+
+            // get axes with sliding window dimensions
+            var axes = this.chart.axesList.filter(function(axis) {
+                var dim = axis.role.grouping.firstDimension;
+                return dim.name == this.dimName;
+            },this);
+
+            axes.forEach(function(axis) {
+                if(axis.option.isDefined('FixedLength')){
+                    if(this.option.isSpecified('Interval')) axis.setInitialLength(this.interval);    //review
+                } 
+                if((axis.option.isDefined('FixedLength')       && 
+                    axis.option.isDefined('PreserveRatio'))    &&
+                    this.option.isSpecified('Interval')        &&  //review
+                            !(axis.option.isSpecified('Ratio')        || 
+                              axis.option.isSpecified('PreserveRatio'))) {
+                    axis.option.specify({ PreserveRatio : true });
+                }
+
+            },this);
+
+        }
+
+    },
+
+
+  options: {
+
+        Interval: {
+            resolve: '_resolveFull',
+            cast: slidingWindow_castInterval,
+            value: Number.MAX_VALUE
+        },
+
+        DimName:   {
+            resolve: '_resolveFull',
+            data: {
+                resolveDefault: function(optionInfo) {
+                    var dv = _defaultDimensionName(this.chart);
+                    optionInfo.defaultValue(dv);
+                    return true;
+                }
+            },
+            cast: slidingWindow_castDimName
+        },
+
+        Score: {
+            resolve: '_resolveFull',
+            data: {
+                resolveDefault: function(optionInfo) {
+                    optionInfo.defaultValue(this._defaultSlidingWindowScore);
+                    return true;
+                }
+            },
+        },
+
+        Select: {
+            resolve: '_resolveFull',
+            data: {
+                resolveDefault: function(optionInfo) {
+                    optionInfo.defaultValue(this._defaultSlidingWindowSelect);
+                    return true;
+                }
+            },
+        }
+
+    }
+
+
+}));
+
+
+function _defaultDimensionName(chart) {
+
+    var dims, dimName;
+    if(!! chart.axes.base) dimName = chart.axes.base.role.grouping.lastDimensionName(); //cart charts always have a base and ortho axis
```

Comment:
**Formatting**
1. Don´t use unbalanced if/else parentheses. If one of the branches requires parentheses, then put parentheses on both.
2. Don't put a space between an unary operator and the operand, like `!`.
3. Should have a space after the else: `else {`